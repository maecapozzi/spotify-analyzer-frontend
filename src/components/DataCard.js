import React from 'react'
import injectSheet from 'react-jss'
import PropTypes from 'prop-types'

import Card from 'material-ui/Card'

const styles = {
  card: {
    border: '1px #272B2D solid',
    padding: '2em',
    backgroundColor: '#181A1B !important',
    height: '430px'
  }
}

const DataCard = ({ classes, component }) => {
  return <Card className={classes.card}>{component}</Card>
}

DataCard.propTypes = {
  component: PropTypes.object
}
export default injectSheet(styles)(DataCard)
