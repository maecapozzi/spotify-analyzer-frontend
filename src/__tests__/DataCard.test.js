import React from 'react'
import { shallow } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'

import DataCard from '../components/DataCard'

configure({ adapter: new Adapter() })

test('it renders correctly', () => {
  const component = shallow(<DataCard />)
  expect(component).toMatchSnapshot()
})
