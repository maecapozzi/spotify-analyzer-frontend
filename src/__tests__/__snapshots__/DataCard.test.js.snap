// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(DataCard) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "card": "DataCard-card-0-1",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "card": "DataCard-card-0-1",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
